### PR TITLE
Wait for localhost kas to respond

### PR DESCRIPTION
**Pull request by soltysh**
Similar to https://github.com/openshift/cluster-kube-controller-manager-operator/pull/420

/assign @deads2k

### CODE REVIEW

**Comment by deads2k** [comment]
/lgtm

but I'm pretty sure we broke logging, didn't we....

**Comment by deads2k** [comment]
/hold

holding so you can fix the logging aspect.  I recommend using substitution like we did here https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/targetconfigcontroller/targetconfigcontroller.go#L205

**Comment by soltysh** [comment]
/hold cancel 
the 2nd commit should fix it

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -299,33 +299,43 @@ func managePod_v311_00_to_latest(ctx context.Context, configMapsGetter corev1cli
...
 
 	if _, err := secretsGetter.Secrets(required.Namespace).Get(ctx, "serving-cert", metav1.GetOptions{}); err != nil && !apierrors.IsNotFound(err) {
 		return nil, false, err
 	} else if err == nil {
-		required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, "--tls-cert-file=/etc/kubernetes/static-pod-resources/secrets/serving-cert/tls.crt")
-		required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, "--tls-private-key-file=/etc/kubernetes/static-pod-resources/secrets/serving-cert/tls.key")
+		containerArgsWithLoglevel[0] += " --tls-cert-file=/etc/kubernetes/static-pod-resources/secrets/serving-cert/tls.crt"
+		containerArgsWithLoglevel[0] += " --tls-private-key-file=/etc/kubernetes/static-pod-resources/secrets/serving-cert/tls.key"
 	}
+	containerArgsWithLoglevel[0] = strings.TrimSpace(containerArgsWithLoglevel[0])
```
Comment (deads2k) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:338]: don't you have to assign this?

**Comment by damemi** [comment]
@soltysh looks like you're getting a crashloop in the kube-scheduler-operator pod:
```
level=error msg="Cluster operator kube-scheduler Degraded is True with NodeInstaller_InstallerPodFailed::StaticPods_Error: NodeInstallerDegraded: 1 nodes are failing on revision 4:\nNodeInstallerDegraded: ; 1 nodes are failing on revision 5:\nNodeInstallerDegraded: static pod of revision 5 has been installed, but is not ready while new revision 6 is pending; 1 nodes are failing on revision 6:\nNodeInstallerDegraded: static pod of revision 6 has been installed, but is not ready while new revision 7 is pending\nStaticPodsDegraded: pod/openshift-kube-scheduler-ip-10-0-150-179.us-east-2.compute.internal container \"kube-scheduler\" is not ready: CrashLoopBackOff: back-off 5m0s restarting failed container=kube-scheduler pod=openshift-kube-scheduler-ip-10-0-150-179.us-east-2.compute.internal_openshift-kube-scheduler(967b4de0af1a290ade33d846524cb6ef)\nStaticPodsDegraded: pod/openshift-kube-scheduler-ip-10-0-150-179.us-east-2.compute.internal container \"kube-scheduler\" is waiting: CrashLoopBackOff: back-off 5m0s restarting failed container=kube-scheduler pod=openshift-kube-scheduler-ip-10-0-150-179.us-east-2.compute.internal_openshift-kube-scheduler(967b4de0af1a290ade33d846524cb6ef)\nStaticPodsDegraded: pod/openshift-kube-scheduler-ip-10-0-216-190.us-east-2.compute.internal container \"kube-scheduler\" is not ready: CrashLoopBackOff: back-off 5m0s restarting failed container=kube-scheduler pod=openshift-kube-scheduler-ip-10-0-216-190.us-east-2.compute.internal_openshift-kube-scheduler(202232afd51dcd271171a0b6a88af98b)\nStaticPodsDegraded: pod/openshift-kube-scheduler-ip-10-0-216-190.us-east-2.compute.internal container \"kube-scheduler\" is waiting: CrashLoopBackOff: back-off 5m0s restarting failed container=kube-scheduler pod=openshift-kube-scheduler-ip-10-0-216-190.us-east-2.compute.internal_openshift-kube-scheduler(202232afd51dcd271171a0b6a88af98b)\nStaticPodsDegraded: pod/openshift-kube-scheduler-ip-10-0-147-222.us-east-2.compute.internal container \"kube-scheduler\" is not ready: CrashLoopBackOff: back-off 5m0s restarting failed container=kube-scheduler pod=openshift-kube-scheduler-ip-10-0-147-222.us-east-2.compute.internal_openshift-kube-scheduler(16a567c312ffe56eae385063cb847af4)\nStaticPodsDegraded: pod/openshift-kube-scheduler-ip-10-0-147-222.us-east-2.compute.internal container \"kube-scheduler\" is waiting: CrashLoopBackOff: back-off 5m0s restarting failed container=kube-scheduler pod=openshift-kube-scheduler-ip-10-0-147-222.us-east-2.compute.internal_openshift-kube-scheduler(16a567c312ffe56eae385063cb847af4)"
```

**Comment by damemi** [comment]
Has this been tested manually in a cluster?

**Comment by damemi** [comment]
`level=info msg="Cluster operator kube-scheduler Available is False with StaticPods_ZeroNodesActive: StaticPodsAvailable: 0 nodes are active; 3 nodes are at revision 0; 0 nodes have achieved new revision 7"` in e2e-aws too
/lgtm cancel
to sort out tests

**Comment by soltysh** [comment]
There's a debate between @mfojtik and @tnozicka about just this problem, I'd hold with adding it until we sort it out. We'll be talking about it tomorrow during arch call.

**Pull request closed without merging**